tools/internal_ci/linux/grpc_e2e_performance_gke.sh

```diff
@@ -55,7 +58,7 @@ WORKER_POOL_8CORE=workers-8core-ci
 WORKER_POOL_32CORE=workers-32core-ci
 
 # Update go version.
-TEST_INFRA_GOVERSION=go1.17
+TEST_INFRA_GOVERSION="$(curl https://golang.org/VERSION?m=text)"
```

Comment:
let's keep the go version pinned. Targeting a version that's changing under our hands can lead to sudden breakages. 
Fetching the latest version is also one more http request (that can fail and cause unnecessary flakiness).

also below.